addons/account/models/account_move.py

```diff
@@ -5586,7 +5586,10 @@ def _get_invoice_next_payment_values(self, custom_amount=None):
             next_due_date = epd_installment['date_maturity']
             discount_date = epd_installment['line'].discount_date
```

Comment:
```suggestion
            discount_date = epd_installment['line'].discount_date or fields.Date.context_today(self)
```
We should take today's date as discount date if not provided